frontend/src/components/common/CodeEditor.jsx

```
@@ -0,0 +1,255 @@
+  initialValue,
+  onChange,
+  onRun,
+  onError,
+  markers = [],
+  options = {}
+}) {
+  const { language } = useLanguage();
+  const editorRef = useRef(null);
+  const monacoRef = useRef(null);
+  const [isDarkMode, setIsDarkMode] = useState(
+    window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)